### DISCUSSION TITLE
Can't reproduce VarCouldBeVal and UnreachableCode

### DISCUSSION THREAD

**Discussion by GirZ0n** [detekt/detekt#4077] [Q&A] [answered]
I try to reproduce [`VarCouldBeVal`](https://detekt.github.io/detekt/style.html#varcouldbeval) and [`UnreachableCode`](https://detekt.github.io/detekt/potential-bugs.html#unreachablecode) using examples from the documentation, but for some reason, Detekt does not find these errors.

I run Detekt with the following command: `java -jar detekt-cli-1.18.0-all.jar --build-upon-default-config --input test.kt`

```kotlin
// test.kt

fun test() {
    var a = 1 // violation: this variable is never re-assigned
    val b = a + 2
}

fun main() {
    throw IllegalArgumentException()
    println() // unreachable
}
```

Output:
```
exceptions - 5min debt
	ThrowingExceptionsWithoutMessageOrCause - [main] at /home/ilya/Downloads/test.kt:7:11
style - 5min debt
	UnusedPrivateMember - [b] at /home/ilya/Downloads/test.kt:3:5

Overall debt: 10min

Build failed with 2 weighted issues.
```

What am I doing wrong?

**Comment by marschwar** [comment, marked as answer]
The rule `VarCouldBeVal` requires type resolution. Please refer to to [documentation](https://detekt.github.io/detekt/type-resolution.html) on how to set up your environment in order for this rule to work.

**Reply by GirZ0n** [reply]
Thank you! It works now.
